engine/lib/elgglib.php

```diff
@@ -1426,7 +1426,31 @@ function _elgg_normalize_plural_options_array($options, $singulars) {
  */
 function _elgg_shutdown_hook() {
 	global $START_MICROTIME;
+	
+	elgg_set_config('shutdown_flag', true);
```

Comment:
the config flag is useful when you have something triggering on an event.

eg. when someone joins a group -> do something
if the 'do something' is time consuming you might want to defer it to after shutdown, but if they joined the group as part of a shutdown script then you can't defer it.

pseudocode is

    if (elgg_get_config('shutdown_flag')) {
        // do something now
    }
    else {
        // defer to after shutdown
    }